subprojects/provider/src/test/kotlin/org/gradle/kotlin/dsl/accessors/ProjectSchemaTest.kt

```diff
@@ -199,21 +203,132 @@ class ProjectSchemaTest : TestWithClassPath() {
         )
     }
 
+    @Test
+    fun `cache key takes configurations into account`() {
+
+        assertThat(
+            cacheKeyFor(
+                configurations = listOf("api")
+            ),
+            not(equalTo(cacheKeyFor(
+                configurations = listOf("implementation")
+            )))
+        )
+    }
+
+    @Test
+    fun `cache key takes container element names into account`() {
+
+        assertThat(
+            cacheKeyFor(
+                containerElements = listOf(
+                    entry<TaskContainer, DefaultTask>("assemble")
+                )
+            ),
+            not(equalTo(cacheKeyFor(
+                containerElements = listOf(
+                    entry<TaskContainer, DefaultTask>("clean")
+                )
+            )))
+        )
+    }
+
+    @Test
+    fun `cache key takes container element types into account`() {
+
+        assertThat(
+            cacheKeyFor(
+                containerElements = listOf(
+                    entry<TaskContainer, DefaultTask>("clean")
+                )
+            ),
+            not(equalTo(cacheKeyFor(
+                containerElements = listOf(
+                    entry<TaskContainer, Delete>("clean")
+                )
+            )))
+        )
+    }
+
+    @Test
+    fun `cache key takes task names into account`() {
+
+        assertThat(
+            cacheKeyFor(
+                tasks = listOf(
+                    entry<TaskContainer, DefaultTask>("assemble")
+                )
+            ),
+            not(equalTo(cacheKeyFor(
+                tasks = listOf(
+                    entry<TaskContainer, DefaultTask>("clean")
+                )
+            )))
+        )
+    }
+
+    @Test
+    fun `cache key takes task types into account`() {
+
+        assertThat(
+            cacheKeyFor(
+                tasks = listOf(
+                    entry<TaskContainer, DefaultTask>("clean")
+                )
+            ),
+            not(equalTo(cacheKeyFor(
+                tasks = listOf(
+                    entry<TaskContainer, Delete>("clean")
+                )
+            )))
+        )
+    }
```

Comment:
All of these assertions assert the lack of equality; any value in adding a test that covers checking that there is a case where equality is equal?
Is this already handled by `public type from jar is represented as Accessible`?

Reply:
We do have integration tests for caching which indirectly test the equality of cache keys but it makes sense to add a few unit tests as well. Done.